lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -2547,6 +2548,26 @@ def _add_image_features(self, node_space: schema.NodeSpace) -> None:
         else:
             ...
 
+    def _check_image_capability(self, node_space: schema.NodeSpace) -> None:
+        is_an_supported: Optional[bool] = None
+        azure_runbook = node_space.get_extended_runbook(AzureNodeSchema, AZURE)
+        if azure_runbook.vhd:
+            is_an_supported = azure_runbook.vhd.is_accelerated_network_supported
+
+        # Network interface feature is handled by node_space directly.
+        # Don't need to add in featuers
+        if isinstance(is_an_supported, bool) and not is_an_supported:
+            data_path = schema.NetworkDataPath.Synthetic
+            # Check if requirement is supported in image capability
+            if node_space.network_interface:
```

Comment:
I think it doesn't need to call `check_xxx`, just set it to Synthetic.

Reply:
Some test cases require SRIOV data_path, such as verify_sriov_basic, then node_space.network_interface.data_path is SRIOV. For these cases, they should raise NotMeetRequirementException if VHD doesn't support AN before deploying the VM.

If just set node_space.network_interface.data_path to Synthetic, the logic can still find a matched VM size and deploy the VM. Until running the case, it just skips the case.

So it still needs to cal check_xxx, if not match, then raise NotMeetRequirementException before copying VHD and deploying.

Reply:
Does the image work with AN VM sizes, but AN is disabled? Or it needs the AN unavailable vm sizes?

Reply:
Yes, the image works with AN VM sizes but AN is disabled.